app/src/main/java/woowacourse/omok/MainActivity.kt

```diff
@@ -0,0 +1,176 @@
+                }
+            }
+    }
+
+    // 디비에 저장된 돌 상태들을 들고 온다
+    private fun initGame() {
+        dbProvider.createTable()
+        val stoneState = dbProvider.readAll()
+        stoneState.forEach { stone ->
+            omokGame.grid.putStone(stone)
+        }
+    }
+
+    // 턴을 진행한다
+    private fun proceedTurn(
+        view: ImageView,
+        stoneColor: StoneColor,
+    ): StoneColor {
+        val point = Stone(view.tag as Point, stoneColor)
+        if (isViolation(stoneColor, point)) return stoneColor
+        playMove(stoneColor, view)
+        if (checkGameOver(stoneColor, point)) completeGame(stoneColor, point)
+        return omokGame.changeTurn(stoneColor)
+    }
+
+    // 둘 수 있는 위치인지 확인한다
+    private fun isViolation(
+        stoneColor: StoneColor,
+        point: Stone,
+    ): Boolean {
+        val result = omokGame.validatePoint(stoneColor, point)
+        if (result is ValidationResult.Success) return false
+        showViolation(result as ValidationResult.Failure)
+        return true
+    }
+
+    // 착수한다
+    private fun playMove(
+        stoneColor: StoneColor,
+        view: ImageView,
+    ) {
+        view.setImageResource(getStoneImage(stoneColor))
+        omokGame.playMove(Stone(view.tag as Point, stoneColor))
+        dbProvider.insertStone(Stone(view.tag as Point, stoneColor))
+    }
+
+    // 돌 색깔별로 이미지를 받아온다
+    private fun getStoneImage(stoneColor: StoneColor): Int {
+        return when (stoneColor) {
+            StoneColor.BLACK -> R.drawable.black_stone
+            StoneColor.WHITE -> R.drawable.white_stone
+        }
+    }
+
+    // 게임이 끝났는지 확인한다
+    private fun checkGameOver(
+        stoneColor: StoneColor,
+        point: Stone,
+    ): Boolean {
+        return omokGame.checkWin(stoneColor, point) || omokGame.isBoardFull()
+    }
+
+    // 게임을 끝낸다
+    private fun completeGame(
+        stoneColor: StoneColor,
+        point: Stone,
+    ) {
+        showGameResult(stoneColor, point)
+        isGameOver = true
+        dbProvider.dropTable()
+    }
+
+    // 게임 결과를 출력한다
+    private fun showGameResult(
+        stoneColor: StoneColor,
+        point: Stone,
+    ) {
+        when {
+            omokGame.checkWin(stoneColor, point) -> {
+                showToast(MESSAGE_WINNER.format(OmokResult.getWinner(stoneColor)))
+            }
+
+            omokGame.isBoardFull() -> {
+                showToast(MESSAGE_WINNER.format(OmokResult.DRAW))
+            }
+        }
+    }
+
+    private fun showViolation(violation: ValidationResult.Failure) {
+        val errorText =
+            when (violation) {
+                ValidationResult.Failure.DoubleFour -> getString(R.string.error_double_four)
+                ValidationResult.Failure.DoubleThree -> getString(R.string.error_double_three)
+                ValidationResult.Failure.Occupied -> getString(R.string.error_occupied)
+                ValidationResult.Failure.OverLine -> getString(R.string.error_over_line)
+            }
```

Comment:
`when`, sealed 클래스 활용 + string을 가져오는 방법 👍

```suggestion
import woowacourse.omok.domain.rule.ValidationResult.Success
import woowacourse.omok.domain.rule.ValidationResult.Failure
            when (violation) {
                Failure.DoubleFour -> getString(R.string.error_double_four)
                Failure.DoubleThree -> getString(R.string.error_double_three)
                Failure.Occupied -> getString(R.string.error_occupied)
                Failure.OverLine -> getString(R.string.error_over_line)
            }
```
`import`를 수정해서 이렇게 작성해볼 수도 있겠습니다.

Reply:
[해당 내용 관련 커밋입니다](https://github.com/woowacourse/kotlin-omok/pull/128/commits/74f23386fb4413efd4b59542f84623c09aa8345e)